services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java

```diff
@@ -110,8 +110,15 @@ public int postText(String uri, String text) throws IOException {
             public InputStream get(String uri) throws IOException {
                 HttpGet request = new HttpGet(uri);
                 HttpResponse response = httpClient.execute(request);
+                int statusCode = response.getStatusLine().getStatusCode();
                 HttpEntity entity = response.getEntity();
-                if (entity == null) return null;
+                if (statusCode < 200 || statusCode >= 300) {
```

Comment:
One note: This won't handle redirections. Is it on purpose? Or is it any issue with switching the latter part to `statusCode >= 400)` ?

I wonder about scenario like:
- Well-known endpoint configured for some IDP to `https://my-idp-public-endpoint`
- This will redirect to something like `https://my-real-endpoint` 

On the other hand, HttpClient should do some redirections automatically, so maybe this is not an issue?

Reply:
@mposolda - HttpClient will already to an automatic redirect on GET, see https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/impl/client/DefaultRedirectStrategy.html. If we would end up with a 301 here, the body would be empty anyway, and the current code can't continue. 

The code I added here is similar to https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/client/AbstractResponseHandler.java#L64-L74 which I'll use in a follow-up PR listed above.

Reply:
@ahus1 Yes, That's what I though (about httpClient doing automatic redirection) , but wasn't 100% sure. Thanks for answer this!